### ISSUE TITLE
Confusion around loading a GGUF locally

### ISSUE THREAD

**Issue by mojadem** [EricLBuehler/mistral.rs#922]
In `mistralrs/examples/gguf_locally/main.rs`, the following example is given to load a GGUF locally:

```rust
let model = GgufModelBuilder::new(
    "gguf_models/mistral_v0.1/",
    vec!["mistral-7b-instruct-v0.1.Q4_K_M.gguf"],
)
```

So, the first parameter appears to be the path to the parent directory, and the second is the actual GGUF file. This doesn't seem to correspond with the function signature for `GgufModelBuilder::new`:

```rust
pub fn new(model_id: impl ToString, files: Vec<impl ToString>) -> Self
```

Why is the parent directory used as the `model_id`? Maybe there is room for improving the API for local vs. HF GGUFs, as I understand the `model_id` when using a GGUF from HF refers to the HF repo, which acts well as a `model_id`.

**Comment by kloudsamurai** [comment]
Wouldn't it just be easiest to clone your model into a directory named the same as the hf remote? then you'd have the alignment desired.

**Comment by mojadem** [comment]
Yes, you're right, that would provide the correct file structure. However, to me it would make more sense if we had something like `GgufModelBuilder::new("path/to/model.gguf)` or `GgufModelBuilder::from_file("path/to/model.gguf")`.
